crates/env/src/engine/off_chain/impls.rs

```diff
@@ -226,14 +229,19 @@ impl EnvBackend for EnvInstance {
     where
         T: scale::Decode,
     {
-        unimplemented!("the off-chain env does not implement `seal_input`")
+        T::decode(&mut self.engine.exec_context.borrow().input.as_slice())
+            .map_err(|_| Error::CalleeTrapped)
     }
 
-    fn return_value<R>(&mut self, _flags: ReturnFlags, _return_value: &R) -> !
+    fn return_value<R>(&mut self, flags: ReturnFlags, return_value: &R) -> ReturnType
     where
         R: scale::Encode,
     {
-        unimplemented!("the off-chain env does not implement `seal_return_value`")
+        if flags.is_reverted() {
+            panic!("the off-chain env does not implement revert in `seal_return_value`")
+        }
```

Comment:
Just tested it out in a relatively large project. This is the only problem I've encountered so far. If this one is not implemented, if any function returns an `Err`, the unit test will be interrupted by the `panic`.